core-backup/src/main/scala/io/aiven/guardian/kafka/backup/BackupClientInterface.scala

```diff
@@ -247,18 +271,37 @@ trait BackupClientInterface[T <: KafkaClientInterface] {
     }
   }
 
+  /** Prepares the sink before it gets handed to `backupToStorageSink`
+    */
+  private[backup] def prepareStartOfStream(state: Option[CurrentState],
```

Comment:
This function handles the case where if there is current state (i.e. you are resuming from a previously upload) then we need to drop the first character `[` from the stream since there is an already existing half complete upload which is the middle of an array